react/private/withTextProps.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+
+const sizes = [
+  'small',
+  'substandard',
+  'superstandard',
+  'subheading',
+  'headline',
+  'heading',
+  'hero'
+];
+const getSizeProps = props => {
+  const sizeProps = {};
+
+  sizes.forEach(size => {
+    if (props.size) {
+      sizeProps[size] = props.size === size || props[size];
+    } else {
+      sizeProps.size = props[size] ? size : sizeProps.size;
+    }
+
+    return;
+  });
+
+  return sizeProps;
+};
+
+const withTextProps = () => WrappedComponent => {
+  return props => {
+    const newProps = {
+      ...props,
```

Comment:
We should be omitting boolean size props here before spreading props, otherwise both will get passed down to `WrappedComponent`. We actually see this in the snapshots in this PR, where `size="heading" heading={true}` etc. is coming through.

Reply:
That's by design, that's the idea, so you can use both

Reply:
You should still be able to use both, this decorator just allows you to simplify the decorated component by taking care of boolean props for you. Might pair on this one :)